pkg/foundation/cerrors/fatal.go

```diff
@@ -0,0 +1,48 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cerrors
+
+import (
+	"fmt"
+)
+
+// FatalError is an error type that will differentiate these from other errors that could be retried.
+type FatalError struct {
```

Comment:
It looks like we can make this type unexported, similar to `errors.New` (that returns an `*errorString`) or `fmt.Errorf` (that returns an `*wrapError`).

Reply:
@hariso good point.